### PR TITLE
feat(gatsby-transformer-react-docgen): use local babel config for react-docgen

### PR DESCRIPTION
**Pull request by jquense**
BREAKING CHANGE: now uses the local babel config with a fallback

### CODE REVIEW

**Comment by jquense** [comment]
ping :)

File: packages/gatsby-transformer-react-docgen/src/__tests__/on-node-create.js [hunk 1]
```diff
@@ -114,10 +119,10 @@ describe(`transformer-react-doc-gen: onCreateNode`, () => {
 
   it(`should infer a name`, async () => {
     node.__fixture = `unnamed.js`
-    node.absolutePath = path.join(__dirname, `UnnamedExport`)
+    // node.absolutePath = path.join(__dirname, `UnnamedExport`)
```
Comment (DSchau) [inline packages/gatsby-transformer-react-docgen/src/__tests__/on-node-create.js]: Is there a reason this is commented out?

Comment (DSchau) [reply packages/gatsby-transformer-react-docgen/src/__tests__/on-node-create.js]: Should we just remove?

**Comment by jquense** [comment]
ya i'm exploring what is required from the docs, give me a sec i've pushed a version to a scoped package to try out!

**Comment by jquense** [comment]
Ok needed some additional work donw upstream, but thats merged and in 👍 

> If I'm reading correctly, it looks like we now require a cwd and parserOptions passed to the plugin, but we don't document this case.

Not required, only passed through for the sake the tests so that the packages own `.babelrc` doesn't affect the test outcomes.

I've documented the options as well so it's clearer what the behavior is and how to adjust it.

**Comment by jquense** [comment]
ok good to go 👍

**Comment by jquense** [comment]
friendly ping

**Comment by LekoArts** [comment]
I'm also interested in this update, as I'll then be able to use this resolver, right?
https://github.com/Jmeyering/react-docgen-annotation-resolver

Nudge for @DSchau 	👍

**Comment by jquense** [comment]
@LekoArts you can specify a resolver now, so that should already work.

**Comment by DSchau** [comment]
Waiting for the CI checks to pass--then will get this merged in and published as `gatsby-transformer-react-docgen`@`4.0.0`.

**Comment by DSchau** [comment]
Published as [`gatsby-transformer-react-docgen`@`4.0.0`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-react-docgen/CHANGELOG.md#400-2019-04-02)

**Pull request merged**